pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
hmm, I think preparing API endpoint at "/api/metrics/prometheus" is good way.

> Grafana counters are like statsd counters (reset each interval). And by using pull Prometheus must pull at the exact same rate as Grafana tries to "push", which might cause issues.

I'm not sure, but if we push the metrics, the problem is just move from Grafana to pushgateway. There is same problem.

@brian-brazil If you have a time, would you check this and advice for me?